### ISSUE TITLE
Certificate generator

### ISSUE THREAD

**Issue by hxmelab** [holomekc/bosch-smart-home-bridge#2]
Hi, first of all thank you for this library. I could get a few things running. 

But to generate a certificate you write in your documentation: 
> absolute path to a directory where client certificates are located or will be generated to

For any reason no certificate or a key gets generated at this point.

```

const path = require('path');
const {BoschSmartHomeBridge} = require('bosch-smart-home-bridge');

const caDev = path.resolve('/home/pi/certs');
const bshb = new BoschSmartHomeBridge('192.168.0.10', 'client-id', caDev, new Logger());

```
When I add this line:
`bshb.pairIfNeeded('client-name', 'password');`

I get: 
`info Check if client with identifier: client-id is already paired.`
But the client did not pair nor are certificates present in the 'caDev' directory.

But when I add a GET on rooms instead of the pairing command
`bshb.getBshcClient().getRooms().subscribe(result => {});`

the certificate and key gets created:
```
info certificate for identifier node-red-contrib-bosch not found. Creating new client certificate.
info Generating RSA private key, 2048 bit long modulus
............................................+++++
..+++++
e is 65537 (0x010001)
```

I would expect this behavior when creating a new instance of BoschSmartHomeBridge but not when I try to get the room array. Did I get something wrong here?

**Comment by holomekc** [comment]
Hi. You need to subscribe to pair if needed. Otherwise nothing happens. The log message is confusing there. I will move it. 

When you check the method you will see that it uses getRooms method internally. And you are right. All other methods generates a certificate as well. This is because of the design of the certificatestorage. I will think about that again. A solution could be:

ClientStorage.getClientCertificate
ClientStorage.generateCertificate (only used by pairIfNeeded the )

Maybe i could get rid of the pairIfNeeded method and do that automatically in basically every method. 

I hope this helps you

**Comment by hxmelab** [comment]
Okay, this is working now. Another question regarding this. Is the following correct?

```
bshb.pairIfNeeded('client-name', 'password', delayInMilliSec, timeOutInSec).subscribe(result => {
   //delayInMilliSec - Delay in milliseconds until pairing process starts
  //timeOutInSec - Timeout for request in seconds
});
```

There is one thing I recognized. When providing a wrong password the request should return a 401 unauthorized. But the request do not return anything. Would it be possible to return the statusCode of the pairing request? That would be great! Thx!

**Comment by holomekc** [comment]
Observable has error and complete callback as well:

bshb.pairIfNeeded(...).subscribe(result => {
// everything is ok. Do awesome stuff.
}, err => {
// Something went wrong.
}, complete => {
// Completed in any case.
});

or you can catch errors:
bshb.pairIfNeeded(...).pipe(catchError(err => {
// Catch errors
}).subscribe(result => {
// everything is ok. Do awesome stuff.
});

This should work normally. Did I miss something there? I will do some tests later.

**Comment by holomekc** [comment]
Hi you are right. Information about issues is not that great. I updated a new version 0.1.0 because the response structure is now different. I created a small test as an example. Check the test directory. 

Basically the response looks like this:
{
  incomingMessage: {<Check http.IncomingMessage>}
  parsedResponse: {<Bshc response parsed>}
}

Also I changed the certificate generation. Only pairIfNeeded will generate the client certificate now.

I hope this is more flexible. Feedback would be great.
